x/leverage/keeper/store.go

```diff
@@ -1,78 +1,81 @@
 package keeper
 
 import (
+	sdkmath "cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	gogotypes "github.com/gogo/protobuf/types"
 
 	"github.com/umee-network/umee/v3/x/leverage/types"
 )
 
-// getAdjustedBorrow gets the adjusted amount borrowed by an address in a given denom. Returns zero
-// if no value is found.
-func (k Keeper) getAdjustedBorrow(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Dec {
-	key := types.CreateAdjustedBorrowKey(addr, denom)
-	adjustedAmount := sdk.ZeroDec()
-
+// getStoredDec retrieves an sdk.Dec from the KVStore, or zero if no value is stored.
+// It panics if a stored value fails to unmarshal into a value higher than a specified minimum.
+// Accepts an additional string which should describe the field being retrieved in panic messages.
+func (k Keeper) getStoredDec(ctx sdk.Context, key []byte, minimum sdk.Dec, desc string) sdk.Dec {
```

Comment:
_Note: Best to view `store.go` and `validate.go` in split screen, since git is mangling the unified diff._

To clarify, the panic messages in these getters aren't meant to be encountered in practice (unless we let a  very serious bug through).

What they will do is replace hard-to-trace panic messages like `negative coin amount: -1` which would be encountered downstream of the getter, with more specific messages like `retrieved an invalid amount: -1 is not above the minimum uToken supply of zero`.

Same applies to the improved error messages on setters, and other cases that might otherwise get through (e.g. `InterestScalar` of `0.5` is non-negative but still invalid.

Hopefully the error messages never actually come into play, but the `val.LTE` checks are so computationally cheap that I thought it would be worth it to use the same logic to check every getter/setter, rather than selectively upgrading the ones most likely to need it, but ending up with a less uniform approach.